src/main/java/net/datafaker/LocaleFaker.java

```diff
@@ -0,0 +1,36 @@
+package net.datafaker;
+
+import java.text.DateFormat;
+import java.util.Arrays;
+import java.util.Locale;
+
+public class LocaleFaker extends AbstractProvider {
+
+    private final Locale[] availableLocales = DateFormat.getAvailableLocales();
+
+    protected LocaleFaker(Faker faker) {
+        super(faker);
+    }
+
+    /**
+     *
+     * @return locale in form: "en_US"
+     */
+    public String baseLocale() {
+        final Locale locale = availableLocales[faker.random().nextInt(availableLocales.length)];
+        return locale.toString();
+    }
+
+    /**
+     *
+     * @return locale in form: "English (United States)"
+     */
+    public String displayName() {
+        int randomIndex = faker.random().nextInt(availableLocales.length - 1);
+        Locale locale = Arrays.stream(availableLocales)
+            .filter(lcl -> !lcl.getDisplayName().isEmpty())
```

Comment:
Why do we need to filter it each time during `displayName` call if it could be filtered once during filling of  `availableLocales` field?